### PR TITLE
fluids: Add cosine hill and skewed advection initial conditions

### PR DESCRIPTION
**Pull request by jrwrigh**
Other changes:
- Change `-bubble_type` to `-advection_ic_type`
- Add test for each new IC

cosine hill:
image

skewed advection:
image
Note that the outflow boundary is set to zero

Both of these are useful for stressing out stabilization schemes, particularly the skewed advection problem. I'll have a follow on PR that sets up having different choices for the $\tau$ constant.

### CODE REVIEW

**Comment by jedbrown** [comment]
Why is skewed advection so jagged? I would have expected something that looked like bilinear elements (this could be a rendering artifact).

**Comment by jrwrigh** [comment]
It's just the initial condition on a relatively coarse mesh. You see the same thing in the original SUPG paper (Brooks Hughes 1982):
image

**Comment by jrwrigh** [comment]
Perhaps this smaller warped plot makes it clearer that this is normal: 
image

Note that to get the warp-by-scalar on a plane, I have to do a slice which converts the quads into triangles. Even if I apply the warp-by-scalar to the volume which (supposedly) keeps it as hexes, most of the ramp down elements have a sharp spine rather than the curved corner shape function we'd expect. I'm guessing that's just a rendering shortcut in ParaView rather than trying to render a curved surface:
image

**Review by jedbrown — APPROVED** [review]
Just a couple stray references to `bubble_type` that I think should be updated alongside. Can the test output binary be any smaller?

File: examples/fluids/README.md [hunk 1]
```diff
@@ -441,8 +441,8 @@ For the 3D advection problem, the following additional command-line options are
   - `1E6`
   - `J`
 
-* - `-bubble_type`
-  - `sphere` (3D) or `cylinder` (2D)
+* - `-advection_ic_type`
+  - Initial condition type, from `sphere`, `cylinder`, `cosine_hill`, and `skew`
```
[the remark this answers is not in the document]
Comment (jrwrigh) [reply examples/fluids/README.md:445]: Good catch. `bubble_type` in the shocktube struct isn't doing anything, so I just removed that (and the other unused advection-related members).

**Comment by jedbrown** [comment]
LGTM. I think this is a squash-merge unless you want to squash in your branch with commits that first added the binary files.

**Comment by jrwrigh** [comment]
Ope, forgot that git would store the old binaries. I'll squash it then merge.

**Pull request merged**